azurerm/data_source_user_assigned_identity_test.go

```diff
@@ -0,0 +1,90 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+)
+
+func TestAccDataSourceAzureRMUserAssignedIdentity_basic(t *testing.T) {
+	dataSourceName := "data.azurerm_user_assigned_identity.test"
+	resourceName := "azurerm_user_assigned_identity.test"
+	ri := tf.AccRandTimeInt()
+	rs := acctest.RandString(4)
+
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceAzureRMUserAssignedIdentity_basic(ri, testLocation(), rs),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(dataSourceName, "name", fmt.Sprintf("acctest%s-uai", rs)),
+					resource.TestCheckResourceAttr(dataSourceName, "resource_group_name", fmt.Sprintf("acctest%d-rg", ri)),
+					resource.TestCheckResourceAttr(dataSourceName, "location", azureRMNormalizeLocation(location)),
+					resource.TestMatchResourceAttr(dataSourceName, "principal_id", validate.UUIDRegExp),
+					resource.TestMatchResourceAttr(dataSourceName, "client_id", validate.UUIDRegExp),
+					resource.TestCheckResourceAttr(dataSourceName, "tags.%", "1"),
+					testEqualResourceAttr(dataSourceName, resourceName, "principal_id"),
+					testEqualResourceAttr(dataSourceName, resourceName, "client_id"),
+				),
+			},
+		},
+	})
+}
+
+func testEqualResourceAttr(dataSourceName string, resourceName string, attrName string) resource.TestCheckFunc {
```

Comment:
I noticed that in other data source tests there is no any checks that data source resource contains same information as a resource.
Do you think this is valuable to move this function to some shared place and reuse in other data source tests?